src/lib/lit-extended.ts

```diff
@@ -70,6 +74,21 @@ export const extendedPartCallback =
           return defaultPartCallback(instance, templatePart, node);
         };
 
+export class SpreadPart extends AttributePart {
+  setValue(values: any[], startIndex: number): void {
+    const properties = values[startIndex];
+    Object.keys(properties).forEach((property) => {
+      const value = properties[property];
+      if (property.endsWith('$')) {
+        const attribute = property!.slice(0, -1);
+        this.element.setAttribute(attribute, value);
+      } else {
```

Comment:
This should handle `on-`prefixed names (and `?` suffixed names in the future).

Reply:
how would we like to do that?
"duplicate code" for setAttribute and assign an property is still ok I guess...

but for event "handling" we probably should use the existing EventPart right? howto best do that?

Reply:
Maybe it'd be useful to actually create `Part` instances for every property of the object. If we'd use `this.instance._partCallback` for that, it'd support event handling, property setting and attribute setting out of the box, as well as any future parts that are added like the `?` suffix for flags. In case anyone extends lit-extended with other custom parts, those would be supported automatically as well.

Although it would look kinda weird to create parts with hardcoded `['', '']` as strings. Calling `setValue` like this also looks weird:

```ts
for (const property in properties) {
  // create subpart if needed

  this.subParts[property].setValue([properties[property]], 0);
}
```

but I think the benefit of not duplicating code and supporting any future and external `Part`s is nice to have.

Reply:
I think @bgotink is right about the approach. It'll reuse the logic in the part callback and naturally gain any new capabilities.

I'd rebase on to the shadycss branch to pick up changes to how part callbacks work.